lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -552,6 +555,137 @@ spec = describe "SHELLEY_STAKE_POOLS" $ do
                 , expectField (#direction . #getApiT) (`shouldBe` Outgoing)
                 ]
 
+    describe "STAKE_POOLS_JOIN_UNSIGNED_01" $ do
+        it "Can join a pool that's not retiring" $ \ctx -> do
+            nonRetiredPools <- eventually "One of the pools should retire." $ do
+                response <- listPools ctx arbitraryStake
+
+                verify response [ expectListSize 3 ]
+
+                pure $ getFromResponse Prelude.id response
+
+            let reportError = error $ unlines
+                    [ "Unable to find a non-retiring pool ID."
+                    , "Test cluster pools:"
+                    , unlines (showT <$> Set.toList testClusterPoolIds)
+                    , "Non-retired pools:"
+                    , unlines (show <$> nonRetiredPools)
+                    ]
+
+            let nonRetiringPoolId = (view #id) . fromMaybe reportError
+                    . find (isNothing . getRetirementEpoch)
+                    $ nonRetiredPools
+
+            -- Join Pool
+            w <- fixtureWallet ctx
+            joinStakePoolUnsigned @n @'Shelley ctx w nonRetiringPoolId >>= \o -> do
+                verify o
+                    [ expectResponseCode HTTP.status200
+                    , expectField #inputs (`shouldSatisfy` (not . null))
+                    , expectField #outputs (`shouldSatisfy` (not . null))
+                    , expectField #certificates (`shouldSatisfy` (not . null))
+                    ]
```

Comment:
We could go further in the assertion and make sure that there are 2 certificates, one about joining, and one about the key registration.